src/shared/utilities/collectionUtils.ts

```diff
@@ -160,3 +160,39 @@ export async function take<T>(sequence: AsyncIterable<T>, count: number): Promis
 
     return result
 }
+
+export interface IteratorForAWSCallParams<TRequest, TResponse> {
+    request: TRequest
+    awsCall: (request: TRequest) => Promise<TResponse>
+    nextTokenNames: {
+        request: keyof TRequest
+        response: keyof TResponse
+    }
+}
+
+/**
+ * Generates an iterator representing a paginated AWS call from a request and an AWS SDK call
+ * @param params Iterator params
+ */
+export async function* getIteratorForAWSCall<TRequest, TResponse>(
+    params: IteratorForAWSCallParams<TRequest, TResponse>
+): AsyncGenerator<TResponse, undefined, TResponse | undefined> {
+    let isDone: boolean = false
+    let nextToken: string | undefined = undefined
+
+    while (!isDone) {
+        const response: TResponse = await params.awsCall({
+            ...params.request,
+            [params.nextTokenNames.request]: nextToken,
+        })
+        if (response[params.nextTokenNames.response]) {
+            nextToken = (response[params.nextTokenNames.response] as any) as string
+        } else {
+            nextToken = undefined
+            isDone = true
```

Comment:
don't need `isDone` flag, can just return immediately
```suggestion
            return undefined
```

Reply:
I'll see if I can make this more concise. There's a couple things here:
* We don't want to return undefined at this point since that would mean you're losing this page.
* We do want to terminate the iterator here since otherwise we'll have one more call that returns undefined since the iterator wouldn't terminate until you've passed through the loop once more.